### PR TITLE
[Test Controller] New test controller applied to Network

### PR DESCRIPTION
**Pull request by vladimir-shcherbakov**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Network/Network.Test/ScenarioTests/ExpressRouteCircuitTests.cs [hunk 1]
```diff
@@ -12,93 +12,90 @@
...
-using Microsoft.Azure.ServiceManagement.Common.Models;
+using Microsoft.Azure.Commands.Network.Test.ScenarioTests;
 using Microsoft.WindowsAzure.Commands.ScenarioTest;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Commands.Network.Test.ScenarioTests
 {
-    public class ExpressRouteCircuitTests : Microsoft.WindowsAzure.Commands.Test.Utilities.Common.RMTestBase
+    public class ExpressRouteCircuitTests :NetworkTestRunner
```
Comment (MiYanni) [inline src/Network/Network.Test/ScenarioTests/ExpressRouteCircuitTests.cs]: Missing space after `:`

File: src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt [hunk 2]
```diff
@@ -1,6 +0,0 @@
-ApplicationSecurityGroupTestsGenerated.cs	299A1E5866F1F33032F70699B1B9A1790689CD35
```
Comment (MiYanni) [inline src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt]: Why is this file deleted? They use a generator to create some of their tests. I'm assuming this file is used by their generator.

File: src/Network/Network.Test/Network.Test.csproj [hunk 3]
```diff
@@ -24,7 +24,6 @@
 
   <ItemGroup>
     <None Update="ScenarioTests\Data\*.*" CopyToOutputDirectory="PreserveNewest" />
-    <None Update="ScenarioTests\Generated\*.*" CopyToOutputDirectory="PreserveNewest" />
```
[the remark this answers is not in the document]
Comment (vladimir-shcherbakov) [reply src/Network/Network.Test/Network.Test.csproj]: I agreed these changes with the Network team

File: src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt (hunk 2 above)
Comment (vladimir-shcherbakov) [reply src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt]: I agreed these changes with the Network team

**Comment by vladimir-shcherbakov** [comment]
@MikhailTryakhov We are applying a new test controller - could you take a look?

File: src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                    @"../AzureRM.Storage.ps1",
+                })
+                .WithNewRmModules (helper => new[]
+                {
+                    helper.RMProfileModule,
+                    helper.GetRMModulePath("AzureRM.Monitor.psd1"),
+                    helper.GetRMModulePath("AzureRM.Network.psd1"),
+                    helper.GetRMModulePath("AzureRM.Compute.psd1"),
+                    helper.GetRMModulePath("AzureRM.ContainerInstance.psd1"),
+                    helper.GetRMModulePath("AzureRM.OperationalInsights.psd1"),
```
Comment (EvgenyAgafonchikov) [inline src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs]: Recently we had to [add Storage reference](https://github.com/Azure/azure-powershell/blob/master/src/Network/Network.Test/NetworkResourcesController.cs#L132) to make some tests pass in Live mode. Do we need the same there?

Comment (vladimir-shcherbakov) [reply src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs]: To use Storage basic functionality we add the [script file](https://github.com/Azure/azure-powershell/blob/481c209b06ff0bbdf05c898226d758844b4c16f9/src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs#L20).

**Comment by vladimir-shcherbakov** [comment]
@MiYanni  Are you ready to sign off?

**Comment by vladimir-shcherbakov** [comment]
signing: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign-esrp/120/

File: src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs (hunk 4 above)
Comment (EvgenyAgafonchikov) [reply src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs]: @vladimir-shcherbakov, I've tried to run tests on this branch (Record mode) and it looks like the test using storage fails. However in case I add AzureRM.Storage.psd1 to this list, it still fails with the same error:
```
  <Results>
    <UnitTestResult executionId="6ed3ba66-daaa-4eda-aa0c-b6ca361696e2" testId="b28595f8-3a5e-64f2-d7eb-8f54b5c7b3f4" testName="Commands.Network.Test.ScenarioTests.NetworkWatcherAPITests.TestTroubleshoot" computerName="EA" duration="00:31:32.2290000" startTime="2019-01-30T14:24:17.4768810+03:00" endTime="2019-01-30T14:24:17.4768815+03:00" testType="13cdc9d9-ddb5-4fa4-a97d-d965ccfc6d4b" outcome="Failed" testListId="8c84fa94-04c1-424b-9868-57a2d4851a1d" relativeResultsDirectory="6ed3ba66-daaa-4eda-aa0c-b6ca361696e2">
      <Output>
        <StdOut>Test-Troubleshoot&#xD;
Caught Exception: System.Management.Automation.CommandNotFoundException: The term 'New-AzStorageContext' is not recognized as the name of a cmdlet, function, script file, or operable program.&#xD;
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.&#xD;
```

Should I update something to make this test pass?

Comment (vladimir-shcherbakov) [reply src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs]: @EvgenyAgafonchikov 
Did you try the live tests on the master branch?

**Comment by MiYanni** [comment]
@vladimir-shcherbakov Please reassign to someone else. Thanks.

File: src/Network/Network.Test/ScenarioTests/NetworkWatcherAPITests.cs [hunk 5]
```diff
@@ -12,77 +12,73 @@
...
         [Fact]
         [Trait(Category.RunType, Category.LiveOnly)]
         [Trait(Category.Owner, Category.netanalyticsdev)]
         public void TestFlowLog()
         {
-            NetworkResourcesController.NewInstance.RunPsTest(_logger, "Test-FlowLog");
+            TestRunner.RunTestScript("Test-FlowLog");
         }
 
 #if NETSTANDARD
```
Comment (markcowl) [inline src/Network/Network.Test/ScenarioTests/NetworkWatcherAPITests.cs]: We should remove this ```#if``` - we are awlays in NetStandard

File: src/Network/Network.Test/ScenarioTests/PublicIpPrefixTests.cs [hunk 6]
```diff
@@ -12,38 +12,35 @@
 // limitations under the License.
 // ----------------------------------------------------------------------------------
 
+using Microsoft.Azure.Commands.Network.Test.ScenarioTests;
```
Comment (markcowl) [inline src/Network/Network.Test/ScenarioTests/PublicIpPrefixTests.cs]: Please braing all the usings outside the namespace

File: src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt (hunk 2 above)
Comment (markcowl) [reply src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt]: This captures the commit for the source script they used to generate the tests, so we should leave it, I think

File: src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs (hunk 4 above)
Comment (markcowl) [reply src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs]: We will either need to enhance the storage.ps1 or add the storage library to this solution and include Storage.psd1 here.  Either way, since there is a separate item to enable runnign the unit tests from the branch, we can defer this untill after the controller is replaced.

File: src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt (hunk 2 above)
Comment (vladimir-shcherbakov) [reply src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt]: @EvgenyAgafonchikov  Can you confirm you need the metadata?

Comment (vladimir-shcherbakov) [reply src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt]: I discussed this offline. 
> AFAIK PS team plans to use their own cmdlets generator, so we are dropping support of ours. In this context removing of this file is okay.

File: src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs (hunk 4 above)
Comment (vladimir-shcherbakov) [reply src/Network/Network.Test/ScenarioTests/NetworkTestRunner.cs]: I discussed this offline with @EvgenyAgafonchikov : 
> Completed checking: test passes after update like this (fail w/o it)
helper.GetRMModulePath("AzureRM.Storage.psd1"),

File: src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt (hunk 2 above)
Comment (markcowl) [reply src/Network/Network.Test/ScenarioTests/Generated/_metadata.txt]: That doesn't apply until we are actually using a new generator.
